### DISCUSSION TITLE
Is there a way to label figures outside the coding chunks that are produced?

### DISCUSSION THREAD

**Discussion by msh855** [quarto-dev/quarto-cli#5216] [Q&A] [answered]
Hi, 

is there a way to label figures outside the coding chunks that are produced? either by R or Python? 

i am using a `qmd` file where 

i have:
```

I want to reference this Figure 

```{python}
# Plot
plot1 = data[['Proxy_rate', 'Fed_rate']].plot(title='The Fed and Forward Guidance')
plt.legend(['Proxy Rate', 'Fund rate'])
plt.show()
```

```

I want to reference in the text the figure produced by that code

Thanks

**Comment by cscheid** [comment]
````
```{python}
#| label: fig-1
plot1 = data[['Proxy_rate', 'Fed_rate']].plot(title='The Fed and Forward Guidance')
plt.legend(['Proxy Rate', 'Fund rate'])
plt.show()
```

This is @fig-1
````

**Reply by msh855** [reply]
excellent, but I get this 
![image](https://user-images.githubusercontent.com/114295360/232571726-5f299d18-1d1b-4648-93db-187c7c4165b6.png)

the question mark and that  In need a caption ? how can I a) switch-off or b) write a caption ? 

I am also interest in having the label centered

**Reply by cscheid** [reply]
https://quarto.org/docs/authoring/figures.html#figure-basics

**Reply by msh855** [reply]
Thanks, still not clear how I can remove completely captions (e.g as a global setting) and how to centre the text of the figure

**Reply by cscheid** [reply]
As of today, you can't have a quarto cross-referenceable figure without an accompanying caption.

**Reply by msh855** [reply]
what about moving the text to the centre?

**Reply by cscheid** [reply]
I don't understand - if there's no caption, what text is there to move? (But, more generally, you'll need to use CSS for that. We don't center captions by design, because captions tend to have longer text and that doesn't look great centered.)

**Reply by msh855** [reply]
![image](https://user-images.githubusercontent.com/114295360/232606694-69c4efe8-0a4c-4f5a-baef-68c49f5c8145.png)

This text (in red) to move to the centre (even if this needs to include a caption)

**Reply by cscheid** [reply]
As I answered above:

> (But, more generally, you'll need to use CSS for that. We don't center captions by design, because captions tend to have longer text and that doesn't look great centered.)

**Reply by mcanouil** [reply]
@msh855 you can look through GitHub Discussions for many examples of using CSS to do many things, such as #2304.

**Reply by msh855** [reply]
Can you please help me with the below? I get this pic

![image](https://user-images.githubusercontent.com/114295360/232853042-03c68f6b-2da2-4378-8805-ef80db699875.png)

Here is the code generating the plot


```{python}
#| label: fig-4

plt3 = df_zcore[['Forward_Guidance_zcore']].plot(color='teal', title='Forward Guidance (zscore)')
plt3.axhline(y=int_range, color='blue', linewidth=0.5, linestyle="--")
plt3.axhline(y=0, color='black', linewidth=1, linestyle="--")
plt3.axhline(y=int_range * -1, color='blue', linewidth=0.5, linestyle="--")

```

why am I getting the things I cycled in the figure? How can I remove them?

**Reply by mcanouil** [reply]
For detailed answer/solution, we[^1] need a detailed problem, in this case we don't need a snippet of code, we need a complete (and small) Quarto document that we can render and work on.

[^1]: "we" is for anyone wanting to help.

**Reply by msh855** [reply, marked as answer]
solved my issue. It turned out, one needs the `plt.show()` at the end of the code with the figure.
